### PR TITLE
query channels

### PR DESCRIPTION
**Pull request by hugurp**
- hooks for data GET
- refactor crud channel for POST and PUT

how do these pattern to contact the api look to you?

### CODE REVIEW

File: src/hooks/use-session.js [hunk 1]
```diff
@@ -1,16 +1,14 @@
 import {useState, useEffect} from 'react'
-import {supabase} from '../utils/supabase-client'
 
-export default function useSession () {
+export default function useSession (sessionProvider) {
```
Comment (oskarrough) [inline src/hooks/use-session.js:3]: how about naming it `database` to stay consistent with the `DbSession` context?

File: src/hooks/use-user-channels.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		}
+		fetchData()
+	}, [userId, database])
+
+	return channels
+}
+
+export {
+	useUserChannels
+}
```
Comment (oskarrough) [inline src/hooks/use-user-channels.js:28]:
```suggestion
export default useUserChannels
```

to make it easier to import and like the other ones

File: src/pages/account.js [hunk 3]
```diff
@@ -1,28 +1,59 @@
...
+} from '../utils/crud/channel'
+import {
+	CreateForm,
+	UpdateForm,
+	DeleteForm
+} from '../components/channel-forms'
 
+export default function Account({dbSession}) {
+	const {session, database} = dbSession
+	const userId = session.user.id
```
Comment (oskarrough) [inline src/pages/account.js:15]:
```suggestion
	const userId = database.auth.user().id
```

or am I missing something?

File: src/pages/account.js [hunk 4]
```diff
@@ -1,28 +1,59 @@
...
-			<h1>R4</h1>
-			<EditForm
-				channel={channels[0]}
-				onEdit={(updates) => mutate([updates])}
-				onDelete={() => mutate([])}
-			></EditForm>
-		</article>
+		<main>
+			<section>
+				{channels && channels.length ? (
```
Comment (oskarrough) [inline src/pages/account.js:20]:
```suggestion
				{channels?.length ? (
```

File: src/utils/crud/channel.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+const createChannel = async ({database, channel, user}) => {
+	const {name, slug} = channel
+	const {id: user_id} = user
+	const now = new Date()
+
+	return await database
```
Comment (oskarrough) [inline src/utils/crud/channel.js:6]: except inside try/catch, there's no need to `return await`. AFAIK should be

```suggestion
	return database
```

See https://jakearchibald.com/2017/await-vs-return-vs-return-await/

File: src/utils/crud/channel.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+			updated_at: new Date()
+		}).eq('id', id).single()
+}
+
+const deleteChannel = async ({database, channel}) => {
+	const {id} = channel
+	console.log(channel)
+	if (!id) return
+
+	return await database
```
Comment (oskarrough) [inline src/utils/crud/channel.js:34]:
```suggestion
	return database
```

**Review by oskarrough — COMMENTED** [review]
Hey cool. I tried to leave a few comments here and there, nothing major. 

What's missing as I see it is a pattern to update the UI, say after creating your channel, updating it etc. What do you say?

File: src/pages/account.js (hunk 3 above)
Comment (hugurp) [reply src/pages/account.js:15]: why change from the `session` to `database`?

File: src/pages/account.js (hunk 4 above)
Comment (hugurp) [reply src/pages/account.js:20]: Yes we could use this syntax, even though it probably "compiles" to the line you changed

File: src/utils/crud/channel.js (hunk 5 above)
Comment (hugurp) [reply src/utils/crud/channel.js:6]: nice, thx for th elink

File: src/hooks/use-session.js (hunk 1 above)
Comment (hugurp) [reply src/hooks/use-session.js:3]: it is not really `database`, as it is `supabase` we are passing.
Supabase in this case , is the "database provider" and "session provider" (as in auth0, and other session as a service providers).

We could rename it to database, but it seems to me we are not interested in making a database query here,
but a "auth" query.

File: src/hooks/use-user-channels.js (hunk 2 above)
Comment (hugurp) [reply src/hooks/use-user-channels.js:28]: yes, it is a legacy change, where i was trying to export all `crud` operations for channels here.
But in the end I thought it was not such a good ideas, as other crud (but read) are not hooks (yet?).

**Pull request merged**